scripts/production/gene_tree_stats.pl

```diff
@@ -0,0 +1,375 @@
+        'Min nb of genes',
+        'Max nb of genes',
+        'Avg nb of species',
+        'Min nb of species',
+        'Max nb of species',
+        'Avg nb of genes per species',
+    ];
+    # Add totals at the end of the array
+    push @tree_sizes_sorted, [
+        undef,
+        'Total',
+        thousandify($sums[0]),
+        thousandify($sums[1]),
+        round2($sums[1] / $sums[0]),
+        $mins[0],
+        thousandify($maxs[0]),
+        round2($sums[2] / $sums[0]),
+        $mins[1],
+        thousandify($maxs[1]),
+        round2($sums[3] / $sums[0]),
+    ];
+
+    return \@tree_sizes_sorted;
+}
+
+
+=head2 get_gene_events
+
+    Calculates the predicted gene events, i.e. for each ancestral species, number of speciation and
+    duplication nodes (including dubious ones), with the average duplication score.
+
+=cut
+
+sub get_gene_events {
+    my $sorted_nodes = shift @_;
+
+    my @gene_events = ();
+    my @sums = (0) x 7;
+    my @tags = qw(nb_nodes nb_dup_nodes nb_gene_splits nb_spec_nodes nb_dubious_nodes avg_dupscore
+                  avg_dupscore_nondub);
+    foreach my $node (@$sorted_nodes) {
+        set_default_value_for_tag($node, 0, @tags);
+        $sums[0] += $node->get_value_for_tag('nb_nodes');
+        $sums[1] += $node->get_value_for_tag('nb_dup_nodes');
+        $sums[2] += $node->get_value_for_tag('nb_gene_splits');
+        $sums[3] += $node->get_value_for_tag('nb_spec_nodes');
+        $sums[4] += $node->get_value_for_tag('nb_dubious_nodes');
+        $sums[5] += $node->get_value_for_tag('avg_dupscore') * ($node->get_value_for_tag('nb_dup_nodes') + $node->get_value_for_tag('nb_dubious_nodes'));
+        $sums[6] += $node->get_value_for_tag('avg_dupscore_nondub') * $node->get_value_for_tag('nb_dup_nodes');
+        push @gene_events, [
+            $node->taxon_id,
+            $node->node_name,
+            thousandify($node->get_value_for_tag('nb_nodes')),
+            thousandify($node->get_value_for_tag('nb_dup_nodes')),
+            thousandify($node->get_value_for_tag('nb_gene_splits')),
+            thousandify($node->get_value_for_tag('nb_spec_nodes')),
+            thousandify($node->get_value_for_tag('nb_dubious_nodes')),
+            ($node->get_value_for_tag('nb_dup_nodes') + $node->get_value_for_tag('nb_dubious_nodes')) ? roundperc2($node->get_value_for_tag('avg_dupscore')) : 'NA',
+            $node->get_value_for_tag('nb_dup_nodes') ? roundperc2($node->get_value_for_tag('avg_dupscore_nondub')) : 'NA',
+        ];
+    }
+    # Sort array by taxon name
+    my @gene_events_sorted = sort { lc($a->[1]) cmp lc($b->[1]) } @gene_events;
+    # Prepend header
+    unshift @gene_events_sorted, [
+        'Taxon ID',
+        'Taxon name',
+        'Nb of nodes',
+        'Nb of duplication nodes',
+        'Nb of gene splits',
+        'Nb of speciation nodes',
+        'Nb of dubious nodes',
+        'Avg confidence score',
+        'Avg confidence score on non-dubious nodes',
```

Comment:
With the values in these columns containing just the numbers, I'd suggest to indicate in the column heading that these are percentages:
```suggestion
        'Avg confidence score (%)',
        'Avg confidence score (%) on non-dubious nodes',
```